spring-cloud-aws-autoconfigure/src/main/java/org/springframework/cloud/aws/autoconfigure/context/ContextCredentialsAutoConfiguration.java

```diff
@@ -32,6 +33,7 @@
  * @author Agim Emruli
  */
 @Configuration
+@ConditionalOnAwsCloudEnvironment
```

Comment:
I am quite new to AWS and have taken a "shotgun" approach here. After a dive through the code I agree this one does not look as though it will explode outside the AWS environment. Removing the condition here does not seem to make my application fail when run locally.